server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamLifecycle.java

```diff
@@ -355,6 +357,17 @@ public static DataStreamLifecycle fromXContent(XContentParser parser) throws IOE
         return PARSER.parse(parser, null);
     }
 
+    /**
+     * Adds a retention param to signal that this serialisation should include the effective retention metadata
+     */
+    public static ToXContent.Params maybeAddEffectiveRetentionParams(ToXContent.Params params) {
+        boolean shouldAddEffectiveRetention = Objects.equals(params.param(RestRequest.PATH_RESTRICTED), "serverless");
+        return new DelegatingMapParams(
```

Comment:
You told me this when we spoke offline and it made sense, but now I can't remember -- why can't we just check PATH_RESTRICTED in `toXContent()`, and not worry about remembering to call `maybeAddEffectiveRetentionParams`?

Reply:
No problem it's a lot to keep in mind. We want only the specific APIs to add the effective retention to their response, at the `DataStreamLifecycle` level we do not know who called us and probably multiple places set the `PATH_RESTRICTED`. So, we are trying to ensure we do not expose this more than we should by explicitly choosing when to set the `maybeAddEffectiveRetentionParams`.

Is this clear now?